src/Action/SetObjectFieldValueAction.php

```diff
@@ -113,8 +113,8 @@ public function __invoke(Request $request): JsonResponse
             $propertyPath = new PropertyPath($field);
         }
 
-        // Handle date and datetime types have setter expecting a DateTime object
-        if ('' !== $value && \in_array($fieldDescription->getType(), ['date', 'datetime'], true)) {
+        // Handle date type has setter expect a DateTime object
+        if ('' !== $value && 'date' === $fieldDescription->getType()) {
```

Comment:
You can use TemplateRegistry::TYPE_DATE here.

In the same way you can use TYPE_BOOLEAN and TYPE_CHOICE for the next lines.

Reply:
Isnt that array change a bc break?

Reply:
He's basically reverting the PR https://github.com/sonata-project/SonataAdminBundle/pull/6071 which is not working for datetime.

Reply:
> You can use TemplateRegistry::TYPE_DATE here.

I did it